velox/exec/ExchangeQueue.cpp

```diff
@@ -63,7 +63,7 @@ std::unique_ptr<SerializedPage> ExchangeQueue::dequeueLocked(
   VELOX_CHECK(future);
   if (!error_.empty()) {
     *atEnd = true;
-    throw std::runtime_error(error_);
+    VELOX_FAIL(error_);
```

Comment:
@amitkdutta plan to store the actual exception here instead of error message to make error message shown on the coordinator side more relevant like the error type. @mbasmanova would you help take care if the followup? Thanks!

Reply:
I see. We would need to change this API then: void ExchangeQueue::setError(const std::string& error)